lib/fog/azurerm/requests/storage/create_block_blob.rb

```diff
@@ -4,8 +4,19 @@ class AzureRM
       # This class provides the actual implementation for service calls.
       class Real
         def create_block_blob(container_name, blob_name, body, options = {})
-          options[:request_id] = SecureRandom.uuid
-          msg = "create_block_blob #{blob_name} to the container #{container_name}. options: #{options}"
+          my_options = options.clone
+          my_options[:request_id] = SecureRandom.uuid
+          if my_options[:create_block_blob_timeout]
```

Comment:
Does options need to have both a `create_block_blob_timeout` value and a `timeout` value? My initial reading of this makes me wonder if it might be clearer/easier if both this and the `put_block_blob_timeout` values were not used and `timeout` was used directly, but I wasn't sure if I was missing something. Does that make sense and/or am I missing something?

Reply:
To (hopefully) clarify, this would then look something like:

```
my_options[:timeout] ||= 656
Fog::Logger.debug "create_block_blob: Setting blob operation timeout to #{my_options[:timeout]} seconds"
```

Note that it would be good to still include the comments about why this is the default, just trying to think if we can cleanup/streamline the code.
```

Reply:
my understanding is that the timeout option has a different semantic outside of the Azure Storage SDK.
If I understand this comment correctly, timeout is for the overall operation while in the Azure Storage SDK, it is for a single put call where one could want to set a different value.

https://github.com/fog/fog-azure-rm/blob/42ca817a35f6a64603ca0a7e27feaf4ad4605f5a/lib/fog/azurerm/models/storage/file.rb#L148

Also, I see it is removed in different places e.g. https://github.com/fog/fog-azure-rm/blob/42ca817a35f6a64603ca0a7e27feaf4ad4605f5a/lib/fog/azurerm/models/storage/file.rb#L155

Does that make sense?

Reply:
Hmm. In those cases, as well as this one, I think the options hash is going to be specific to this call, rather than a global more generic options list though? Or am I misunderstanding/misremembering. ie even if they are named the same thing, their usage would be limited to this narrow context (so it might be ok). Does that make sense?

Reply:
another reason to not use the timeout directly is that the options array is used for 4 different Azure API calls. The function that is blocking in the issue we are trying to solve is this one:
https://github.com/fog/fog-azure-rm/blob/9da5cc04189cd1ec751c5bc1e2f5406ce6b179d0/lib/fog/azurerm/requests/storage/multipart_save_block_blob.rb#L54

The option array is used for:
create_block_blob
put_blob_block
commit_blob_blocks
delete_blob (in the case of an error)

all these Azure API calls upload different sizes (if at all) so they should ideally all have different timeouts. This is not possible if we use the timeout option for all of these.

And I also think the option array is created at the very top. I tried to follow the strack trace and it is even a parameter in this call
https://github.com/fog/fog-azure-rm/blob/9da5cc04189cd1ec751c5bc1e2f5406ce6b179d0/lib/fog/azurerm/models/storage/file.rb#L66

Reply:
Ah ha, yeah I think that gives me the clarity I was lacking. Having more uniquely named variables isn't inherently problematic, but having fewer seems somewhat better if it is possible. It seems like we have a good reason to introduce additional distinct options here though. Thanks for talking through it with me.